commit 5902185c09047644be224e125f1d61717110adb9
Author: Sergei Trofimovich <slyfox@gentoo.org>
Date:   Thu Feb 13 13:34:57 2014 +0300

    mk/config.mk.in: lower -O2 optimization down to -O1 on UNREG
    
    Disable -O2 optimization. Otherwise amount of generated C code makes
    things very slow to compile (~5 minutes on core-i7 for 'compiler/hsSyn/HsExpr.lhs')
    
    And sometimes not compile at all (powerpc64 overflows something
    on 'compiler/hsSyn/HsExpr.lhs').
    
    Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

diff --git a/mk/config.mk.in b/mk/config.mk.in
index b3d6995..6207cce 100644
--- a/mk/config.mk.in
+++ b/mk/config.mk.in
@@ -422,6 +422,16 @@ BIN_DIST_TAR_BZ2      = $(BIN_DIST_NAME)-$(TARGETPLATFORM).tar.bz2
 #
 SRC_HC_OPTS += -H32m -O
 
+# Disable -O2 optimization. Otherwise amount of generated C code
+# makes things very slow to compile (~5 minutes on core-i7 for 'compiler/hsSyn/HsExpr.lhs')
+# and sometimes not compile at all (powerpc64 overflows something
+# on 'compiler/hsSyn/HsExpr.lhs').
+ifeq "$(GhcUnregisterised)" "YES"
+GhcStage1HcOpts=
+GhcStage2HcOpts=
+GhcStage3HcOpts=
+endif
+
 # -----------------------------------------------------------------------------
 # Names of programs in the GHC tree
 
